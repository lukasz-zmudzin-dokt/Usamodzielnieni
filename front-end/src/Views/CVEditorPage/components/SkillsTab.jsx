import React from "react";
import { Button, ButtonToolbar, Form } from "react-bootstrap";
import movie_4 from "../../../assets/movie_4.png";
import Skills from 'Views/CVEditorPage/components/Skills';
import { handleSkillAdd } from 'Views/CVEditorPage/functions/handlers.js';

class SkillsTab extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            skills: [],
            skillToAdd: ""
        }
    }

    render() {
        return (
            <div>
                <Form.Group id="complex-cv-fields">
                    <Form.Label>
                        <h3>Umiejętności</h3>
                        <br></br>
                        <img className="cv_section_img" src={movie_4} width="400vw" alt="" />
                    </Form.Label>
                    <Skills component={this} />
                    <Form.Control
                        className="cv_page_input"
                        type="text"
                        placeholder="Wpisz umiejętność"
                        value={this.state.skillToAdd}
                        onChange={e => {
                            this.setState({ skillToAdd: e.target.value });
                        }}
                    />
                    <Button
                        variant="success"
                        onClick={e => {
                            handleSkillAdd(this, e);
                            console.log(this.state.skills);
                        }}
                    >
                        + Dodaj
                    </Button>
                </Form.Group>
                <ButtonToolbar>
                    <Button
                        className="form_navigation_prev"
                        onClick={e => this.props.onPrevClick(e)}
                    >
                        ← Wstecz
                    </Button>
                    <Button
                        className="form_navigation_next"
                        onClick={e => this.props.onNextClick(e)}
                    >
                        Dalej →
                    </Button>
                </ButtonToolbar>
            </div>
        )
    }
}

export default SkillsTab;
