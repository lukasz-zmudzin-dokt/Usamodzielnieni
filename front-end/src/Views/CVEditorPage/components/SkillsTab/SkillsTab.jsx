import React from "react";
import { Form } from "react-bootstrap";
import movie_4 from "assets/movie_4.png";
import { CVEditorTab, ItemsList } from '../';

class SkillsTab extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            newSkill: {
                name: ''
            }
        }
    }

    getSkill = () => this.state.newSkill;
    getSkillId = (skill) => skill.name;
    getSkillName = (skill) => skill.name;
    clear = () => this.setState({ newSkill: { name: '' } });
    onNameChange = (e) => {
        const name = e.target.value;

        this.setState(prevState => ({
            newSkill: { ...prevState, name }
        }))
    }

    render() {
        return (
            <CVEditorTab
                title="Umiejętności"
                movie={movie_4}
                onPrevClick={this.props.onPrevClick}
                onNextClick={this.props.onNextClick}
                comments={this.props.comments}
                loading={this.props.loading}
                error={this.props.error}
                showComments={this.props.showComments}
            >
                <ItemsList
                    getItemId={this.getSkillId} getItemName={this.getSkillName} getItem={this.getSkill}
                    data={this.props.data} onChange={this.props.onChange} clear={this.clear} refValue={this.props.refValue}
                >
                    <Form.Group controlId="skillName">
                        <Form.Label>Umiejętność</Form.Label>
                        <Form.Control
                            type="text"
                            placeholder="Wpisz umiejętność"
                            value={this.state.newSkill.name}
                            onChange={this.onNameChange}
                        />
                    </Form.Group>
                </ItemsList>
            </CVEditorTab>
        )
    }
}

export default SkillsTab;
