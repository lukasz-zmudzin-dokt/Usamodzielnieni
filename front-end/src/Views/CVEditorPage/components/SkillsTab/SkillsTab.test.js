import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SkillsTab from './SkillsTab';

describe('SkillsTab', () => {
    let props;
    beforeEach(() => {
        props = {
            data: [],
            onChange: () => {}
        }
    });

    it('should render without crashing', () => {
        const { container } = render(
            <SkillsTab {...props} />
        );

        expect(container).toMatchSnapshot();
    });

    it('should call onNameChange function when text input value change', async () => {
        props.onChange = jest.fn();
        const { getByLabelText } = render(
            <SkillsTab {...props} />
        );

        fireEvent.change(
            getByLabelText("Umiejętność", { exact: false }), 
            { target: { value: "Nazwa umiejętności" } }
        );
        expect(getByLabelText("Umiejętność", { exact: false }).value).toBe("Nazwa umiejętności");
    });
});