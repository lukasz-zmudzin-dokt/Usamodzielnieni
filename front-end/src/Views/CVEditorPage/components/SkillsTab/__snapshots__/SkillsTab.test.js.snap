// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsTab should render without crashing 1`] = `
<div>
  <div>
    <h3>
      Umiejętności
    </h3>
    <div
      class="form-group"
    >
      <img
        alt=""
        class="CVEditorTab__img"
        src="movie_4.png"
      />
    </div>
    <div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="skillName"
        >
          Umiejętność
        </label>
        <input
          class="form-control"
          id="skillName"
          placeholder="Wpisz umiejętność"
          type="text"
          value=""
        />
      </div>
      <button
        class="itemsList__addButton btn btn-success"
        type="button"
      >
        + Dodaj
      </button>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6 col-12"
      >
        <button
          class="form_navigation_prev btn btn-primary btn-block"
          disabled=""
          type="button"
        >
          ← Wstecz
        </button>
      </div>
      <div
        class="col-md-6 col-12"
      >
        <button
          class="btn btn-success btn-block"
          id="saveButton"
          type="submit"
        >
          Generuj CV
        </button>
      </div>
    </div>
  </div>
</div>
`;
